### PR TITLE
Build openstack app values from flags

### PR DESCRIPTION
**Pull request by tfussell**
This PR is intended to improve the UX of OpenStack cluster creation. It:
- Drops the `--cluster-app-values` and `--defaults-apps-app-values` flags in favor of building the ConfigMap values entirely dynamically.
- Fetches the latest release of cluster-openstack and default-apps-openstack from the management cluster `AppCatalogEntry` if not specified.
- Adds basic validation for required OpenStack flags.
- Drops `--cluster-topology` flag and non-app-based OpenStack cluster templates.
- Renames `--cluster-app-version` to `--cluster-version`.
- Renames `--cluster-app-catalog` to `--cluster-catalog`.
- Renames `--default-apps-app-version` to `--default-apps-version`.
- Renames `--default-apps-app-catalog` to `--default-apps-catalog`.
- Adds new flags for all important values in https://github.com/giantswarm/cluster-openstack/blob/main/helm/cluster-openstack/values.yaml.
- Drops duplicated provider flags struct in favor of shared structs for passing around values in the provider package.
- Break up `WriteOpenStackTemplateAppCR` into a function for cluster-openstack and a function for default-apps-openstack.

This is not in the changelog as OpenStack is still in alpha and all relevant flags were already hidden.

Example command `kubectl gs template cluster --provider openstack --organization multi-project --enable-oidc --failure-domain test-1 --cloud openstack --cloud-config cloud-config-giantswarm-2 --bastion-image-uuid be59b881-a5a8-4785-85d1-f0c8743bc20d --node-image-uuid 7cd94dd0-2815-427c-ac25-9a1708645dc5 --external-network-id bb5da432-3749-45f5-a1ab-a10eaa5a8a13 --node-cidr 10.6.0.0/24 --bastion-machine-flavor n1.tiny --control-plane-machine-flavor n1.small --worker-machine-flavor n1.small`

Example output
```yaml
---
apiVersion: v1
data:
  values: |
    bastion:
      flavor: n1.tiny
      image: ""
      rootVolume:
        diskSize: 10
        sourceUUID: be59b881-a5a8-4785-85d1-f0c8743bc20d
    cloudConfig: cloud-config-giantswarm-2
    cloudName: openstack
    controlPlane:
      diskSize: 10
      machineFlavor: n1.small
      replicas: 1
    externalNetworkID: bb5da432-3749-45f5-a1ab-a10eaa5a8a13
    nodeCIDR: 10.6.0.0/24
    nodeClasses:
    - diskSize: 10
      machineFlavor: n1.small
      name: default
    nodePools:
    - class: default
      name: default
      replicas: 2
    oidc:
      enabled: true
    organization: multi-project
    rootVolume:
      enabled: true
      sourceUUID: 7cd94dd0-2815-427c-ac25-9a1708645dc5
kind: ConfigMap
metadata:
  creationTimestamp: null
  name: bd00b-userconfig
  namespace: org-multi-project
---
apiVersion: application.giantswarm.io/v1alpha1
kind: App
metadata:
  labels:
    app-operator.giantswarm.io/version: 0.0.0
  name: bd00b
  namespace: org-multi-project
spec:
  catalog: giantswarm
  config:
    configMap:
      name: ""
      namespace: ""
    secret:
      name: ""
      namespace: ""
  kubeConfig:
    context:
      name: ""
    inCluster: true
    secret:
      name: ""
      namespace: ""
  name: cluster-openstack
  namespace: org-multi-project
  userConfig:
    configMap:
      name: bd00b-userconfig
      namespace: org-multi-project
  version: 0.3.0
---
apiVersion: v1
data:
  values: |
    clusterName: bd00b
    oidc:
      enabled: true
    organization: multi-project
kind: ConfigMap
metadata:
  creationTimestamp: null
  name: bd00b-default-apps-userconfig
  namespace: org-multi-project
---
apiVersion: application.giantswarm.io/v1alpha1
kind: App
metadata:
  labels:
    app-operator.giantswarm.io/version: 0.0.0
  name: bd00b-default-apps
  namespace: org-multi-project
spec:
  catalog: giantswarm
  config:
    configMap:
      name: ""
      namespace: ""
    secret:
      name: ""
      namespace: ""
  kubeConfig:
    context:
      name: ""
    inCluster: true
    secret:
      name: ""
      namespace: ""
  name: default-apps-openstack
  namespace: org-multi-project
  userConfig:
    configMap:
      name: bd00b-default-apps-userconfig
      namespace: org-multi-project
  version: 0.1.0
```

### CODE REVIEW

File: cmd/template/cluster/flag.go
Comment (MarcelMue) [inline cmd/template/cluster/flag.go:134]:
```suggestion
	cmd.Flags().StringVar(&f.OpenStack.NodeCIDR, flagOpenStackNodeCIDR, "", "CIDR used for the nodes (OpenStack only).")
```

File: cmd/template/cluster/flag.go
Comment (MarcelMue) [inline cmd/template/cluster/flag.go:123]: I am still a bit uncertain if maintaining all these flags as well as the config file input makes sense. It just seems like a lot of flags to me.

Comment (tfussell) [reply cmd/template/cluster/flag.go:123]: I agree. After using both methods I'm finding the flags more usable. They are documented and discoverable. It avoids extra steps of editing a values file and then running a command. It restricts the interface (can't pass arbitrary values like you can with `values.yaml`) and can be validated.

**Comment by MarcelMue** [comment]
Is this ready for review? Or is something missing before another review?

**Comment by tfussell** [comment]
> Is this ready for review? Or is something missing before another review?

I went down a rabbit hole of trying to fix things. I'll try to get it ready for review today.

**Comment by tfussell** [comment]
Tested cluster creation on `gamma`.

**Review by kopiczko — APPROVED** [review]

**Review by anvddriesch — APPROVED** [review]

**Review by MarcelMue — APPROVED** [review]
This generally is LGTM - not sure if I am missing something as I didn't get to try it much so far.

**Pull request merged**
